server/src/test/java/org/elasticsearch/script/field/ConvertersTestBase.java

```diff
@@ -0,0 +1,509 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.script.JodaCompatibleZonedDateTime;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+/**
+ * A set of the standard available field types for scripting with raw values
+ * contained in a test field to test conversions to other field types.
+ */
+public abstract class ConvertersTestBase extends ESTestCase {
+
+    protected boolean[] rawBooleanValues;
+    protected FieldValues<Boolean> booleanFieldValues;
+    protected Field<Boolean> booleanField;
+
+    @Before
+    public void setupBooleanField() {
+        rawBooleanValues = new boolean[] {
+                true,
+                false
+        };
+
+        booleanFieldValues = new FieldValues<Boolean>() {
+            @Override
+            public boolean isEmpty() {
+                return false;
+            }
+
+            @Override
+            public int size() {
+                return rawBooleanValues.length;
+            }
+
+            @Override
+            public List<Boolean> getValues() {
+                List<Boolean> values = new ArrayList<>();
```

Comment:
Could `rawBooleanValues` change? If not, why not `Arrays.asList(rawBooleanValues)`?

Reply:
Apparently, this is unsupported. I poked around for a while trying to get the correct stream operation.